src/EditorFeatures/CSharpTest/ForeachToFor/ForeachToForTests.cs

```diff
@@ -0,0 +1,1177 @@
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task Comment7()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        // test
+        foreach[||](var a in new int[] { 1, 3, 4 })
+        {
+        }
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        // test
+        var {|Rename:list|} = new int[] { 1, 3, 4 };
+        for (var {|Rename:i|} = 0; i < list.Length; i++)
+        {
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task CommentNotSupported()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach [||] (var a /* test */ in array) ;
+    }
+}
+";
+            await TestMissingInRegularAndScriptAsync(text);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task CommentNotSupported2()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach [||] (/* test */ var a in array) ;
+    }
+}
+";
+            await TestMissingInRegularAndScriptAsync(text);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task CommentNotSupported3()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var array = new int[] { 1, 3, 4 };
+        foreach [||] (var a in array /* test */) ;
+    }
+}
+";
+            await TestMissingInRegularAndScriptAsync(text);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task CollectionStatement()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        foreach[||](var a in new int[] { 1, 3, 4 })
+        {
+            Console.WriteLine(a);
+        }
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        var {|Rename:list|} = new int[] { 1, 3, 4 };
+        for (var {|Rename:i|} = 0; i < list.Length; i++)
+        {
+            var a = list[i];
+            Console.WriteLine(a);
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ForeachToFor)]
+        public async Task CollectionConflict()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var list = 1;
+
+        foreach[||](var a in new int[] { 1, 3, 4 })
+        {
+            Console.WriteLine(a);
+        }
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        var list = 1;
+
+        var {|Rename:list1|} = new int[] { 1, 3, 4 };
```

Comment:
i feel like the name should be "array" not "list".

Reply:
well, I can change it to collection.

Reply:
Please do not.  it is more idiomatic for List/ReadOnlyList to be called "list" and for Array/ImmutableArray to be called "array".

Reply:
do we have name generator that generate proper name?